src/main/java/com/entropyteam/entropay/employees/services/VacationService.java

```diff
@@ -58,7 +63,9 @@ public void addVacationDebit(Employee employee, Integer totalDays) {
                 vacationDebit.setCredit(0);
                 vacationDebit.setDebit(daysToUse);
                 vacationDebit.setEmployee(employee);
-                vacationRepository.save(vacationDebit);
+                Vacation savedEntity = vacationRepository.save(vacationDebit);
+                LOGGER.info("Vacation debit created, employeeId: {}, year: {}, available vacation days before debit: {}, available vacation days after debit: {}",
+                        savedEntity.getEmployee().getId(), savedEntity.getYear(), vacation.getBalance(), vacationRepository.getAvailableDays(employee.getId()));
```

Comment:
Let's remove the `vacationRepository.getAvailableDays(employee.getId())`, it's calling the db in each interation of the for.
The log line should be:
```
LOGGER.info("Vacation debit created, employeeId: {}, year: {}, debit: {}, available vacation days after debit: {}",
                        savedEntity.getEmployee().getId(), savedEntity.getYear(), daysToUse);
```
You could add a log before the for, with the vacations days before debit, reading from `availableVacations.stream().mapToInt(VacationBalanceByYear::getBalance).sum()`